content/blog/rotating-secret-providers/index.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Rotating Secret Providers"
+date: 2020-07-30
+draft: true
+meta_desc: "Pulumi adds ability to rotate stack secret providers"
+meta_image: secrets_rotation.png
+authors:
+    - paul-stack
+
+tags:
+    - Features
+    - Security
+---
+
+One of the features we have had requests from customers and users for is for the ability
+to change the secrets manager associated with their stacks. This would allow a user to rotate
+their secrets providers when people leave their organisation or even to be able to migrate
+to another secret manager of their choice. The [v2.8.0 release](https://github.com/pulumi/pulumi/releases/tag/v2.8.0)
+of Pulumi adds support for this specific feature.
+
+Let's have a look at how to change a secrets provider for an existing stack:
+
+```bash
+pulumi stack change-secrets-provider pulumi stack change-secrets-provider "awskms://alias/ChangeSecretsProvider?region=us-west-2"
+```
+
+This command will change the secrets provider *for the currently selected stack*. It will
+ensure that that the configuration values in `Pulumi.<stack>.yaml` are re-encrypted
+with the new secrets provider and will also ensure that the latest checkpoint (state file)
+also has any secrets held within are migrated with the new secrets provider.
+
+[![asciicast](https://asciinema.org/a/gApPSWG2lNxC4sWxdKIPkuXLc.svg)](https://asciinema.org/a/gApPSWG2lNxC4sWxdKIPkuXLc)
+
+## Creating New Stacks based on existing stacks
+
+When creating a new stack, we also added the ability to create that new stack based on the configuration of
+an existing stack. This means, that when you need to create a new developer stack or a new stack for a
+different environment, then you can create that stack as follows:
+
+```bash
+pulumi stack init test --copy-config-from dev
+```
+
+This will create a new `test` stack and initiate the default secrets provider for the stack. It will then
+get the existing configuration from the `dev` stack and re-encrypt it to the default secrets provider for
```

Comment:
```suggestion
get the existing configuration from the `dev` stack and re-encrypt it to the default secrets provider for
```
get the existing configuration from the dev stack and re-encrypt it to the test stack's default secret provider.